cachedoutput.py

```diff
@@ -41,7 +42,7 @@ def __contains__(self, key):
     def __getitem__(self, key):
         if key not in self:
             raise KeyError(key)
-        with io.open(self._path(key), encoding='utf8') as f:
+        with codecs.open(self._path(key), encoding='utf8') as f:
```

Comment:
This doesn't seem correct. I think `io.open` is the right API to use. Can you provide an example of when `io.open` does the wrong thing?